From 32303900bf390c1af10b185698bf5095677961ff Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Fri, 15 Dec 2023 17:43:45 +0100
Subject: [PATCH 439/841] refactor(st-ddr): reduce error message

Change some very long ERROR messages to VERBOSE, and and a smaller ERROR
message

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 37f1c1df8a0acbf4371097586b5ac5189b4b78a7 refactor(st-ddr): reduce error message]
Change-Id: Ibb205ca9b8f57e3527025e60c2024a041cd906dc
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/347239
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 .../phy/phyinit/src/ddrphy_phyinit_calcmb.c   | 18 +++++++------
 .../phyinit/src/ddrphy_phyinit_f_loaddmem.c   | 10 +++++---
 .../src/ddrphy_phyinit_isdbytedisabled.c      | 25 +++++++++++--------
 .../phyinit/src/ddrphy_phyinit_mapdrvstren.c  |  8 +++---
 .../phyinit/src/ddrphy_phyinit_reginterface.c |  7 +++---
 5 files changed, 42 insertions(+), 26 deletions(-)

diff --git a/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_calcmb.c b/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_calcmb.c
index 5a477e51c..7f8a76da3 100644
--- a/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_calcmb.c
+++ b/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_calcmb.c
@@ -53,19 +53,22 @@ int ddrphy_phyinit_calcmb(void)
 
 	/* A few checks to make sure valid programming */
 	if (nad0 <= 0 || nad1 < 0 || userinputbasic.numdbyte <= 0) {
-		ERROR("%s numactivedbytedfi0, numactivedbytedfi0, NumByte out of range.\n",
+		ERROR("%s %d", __func__, __LINE__);
+		VERBOSE("%s numactivedbytedfi0, numactivedbytedfi0, NumByte out of range.\n",
 		      __func__);
 		return -1;
 	}
 
 	if ((nad0 + nad1) > userinputbasic.numdbyte) {
-		ERROR("%s numactivedbytedfi0+numactivedbytedfi1 is larger than numdbyteDfi0\n",
+		ERROR("%s %d", __func__, __LINE__);
+		VERBOSE("%s numactivedbytedfi0+numactivedbytedfi1 is larger than numdbyteDfi0\n",
 		      __func__);
 		return -1;
 	}
 
 	if (userinputbasic.dfi1exists == 0 && nad1 != 0) {
-		ERROR("%s dfi1exists==0 but numdbyteDfi0 != 0\n", __func__);
+		ERROR("%s %d", __func__, __LINE__);
+		VERBOSE("%s dfi1exists==0 but numdbyteDfi0 != 0\n", __func__);
 		return -1;
 	}
 
@@ -85,10 +88,11 @@ int ddrphy_phyinit_calcmb(void)
 
 #if STM32MP_DDR4_TYPE
 		if (mr4 != 0x0) {
-			ERROR("%s Setting DRAM CAL mode is not supported by the PHY.\n", __func__);
-			ERROR("Memory controller may set CAL mode after PHY has entered mission\n");
-			ERROR("mode. Please check value programmed in mb_ddr_1d[*].mr4\n");
-			ERROR("and unset A8:6\n");
+			ERROR("mr4 != 0x0\n");
+			VERBOSE("%s Setting DRAM CAL mode is not supported by the PHY.\n", __func__);
+			VERBOSE("Memory controller may set CAL mode after PHY has entered mission\n");
+			VERBOSE("mode. Please check value programmed in mb_ddr_1d[*].mr4\n");
+			VERBOSE("and unset A8:6\n");
 			return -1;
 		}
 #endif /* STM32MP_DDR4_TYPE */
diff --git a/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_f_loaddmem.c b/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_f_loaddmem.c
index 5dd486ae8..ff29682c3 100644
--- a/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_f_loaddmem.c
+++ b/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_f_loaddmem.c
@@ -41,12 +41,15 @@ int ddrphy_phyinit_f_loaddmem(int pstate)
 #if STM32MP_DDR3_TYPE || STM32MP_DDR4_TYPE
 	if ((msgblkptr->enableddqs > 8 * (userinputbasic.numactivedbytedfi0)) ||
 	    (msgblkptr->enableddqs <= 0)) {
-		ERROR("%s enableddqs is Zero or greater than NumActiveDbytes for Dfi0\n",__func__);
+		ERROR("%s %d\n", __func__, __LINE__);
+		VERBOSE("%s enableddqs is Zero or greater than NumActiveDbytes for Dfi0\n",
+			__func__);
 		return -1;
 	}
 #elif STM32MP_LPDDR4_TYPE
 	if (msgblkptr->enableddqscha % 16 != 0 || msgblkptr->enableddqschb % 16 != 0) {
-		ERROR("%s Lp3/Lp4 - Number of Dq's Enabled per Channel much be multipe of 16\n",
+		ERROR("%s %d\n", __func__, __LINE__);
+		VERBOSE("%s Lp3/Lp4 - Number of Dq's Enabled per Channel much be multipe of 16\n",
 		      __func__);
 		return -1;
 	}
@@ -54,7 +57,8 @@ int ddrphy_phyinit_f_loaddmem(int pstate)
 	if ((msgblkptr->enableddqscha > 8 * (userinputbasic.numactivedbytedfi0)) ||
 	    (msgblkptr->enableddqschb > 8 * (userinputbasic.numactivedbytedfi1)) ||
 	    (msgblkptr->enableddqscha == 0 && msgblkptr->enableddqschb == 0)) {
-		ERROR("%s EnabledDqsChA/B are not set correctly./1\n", __func__);
+		ERROR("%s %d\n", __func__, __LINE__);
+		VERBOSE("%s EnabledDqsChA/B are not set correctly./1\n", __func__);
 		return -1;
 	}
 #endif /* STM32MP_LPDDR4_TYPE */
diff --git a/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_isdbytedisabled.c b/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_isdbytedisabled.c
index d0ec3a626..9e12c1d3e 100644
--- a/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_isdbytedisabled.c
+++ b/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_isdbytedisabled.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2021-2022, STMicroelectronics - All Rights Reserved
+ * Copyright (C) 2021-2023, STMicroelectronics - All Rights Reserved
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -31,8 +31,9 @@ int ddrphy_phyinit_isdbytedisabled(int dbytenumber)
 	nad1 = userinputbasic.numactivedbytedfi1;
 
 	if (nad0 + nad1 > userinputbasic.numdbyte) {
-		ERROR("%s invalid PHY configuration:\n", __func__);
-		ERROR("numactivedbytedfi0(%d)+numactivedbytedfi1(%d)>numdbytes(%d).\n",
+		ERROR("%s %d", __func__, __LINE__);
+		VERBOSE("%s invalid PHY configuration:\n", __func__);
+		VERBOSE("numactivedbytedfi0(%d)+numactivedbytedfi1(%d)>numdbytes(%d).\n",
 		      nad0, nad1, userinputbasic.numdbyte);
 	}
 
@@ -53,7 +54,8 @@ int ddrphy_phyinit_isdbytedisabled(int dbytenumber)
 	} else if (userinputbasic.dfi1exists == 0x0) {
 		disabledbyte = (dbytenumber > userinputbasic.numactivedbytedfi0 - 1) ? 1 : 0;
 	} else {
-		ERROR("%s invalid PHY configuration:dfi1exists is neither 1 or 0.\n", __func__);
+		ERROR("%s %d", __func__, __LINE__);
+		VERBOSE("%s invalid PHY configuration:dfi1exists is neither 1 or 0.\n", __func__);
 	}
 #endif /* STM32MP_LPDDR4_TYPE */
 
@@ -61,8 +63,9 @@ int ddrphy_phyinit_isdbytedisabled(int dbytenumber)
 #if STM32MP_DDR3_TYPE || STM32MP_DDR4_TYPE
 	if (mb_ddr_1d[0].enableddqs < 1 ||
 	    mb_ddr_1d[0].enableddqs > 8 * userinputbasic.numactivedbytedfi0) {
-		ERROR("%s enableddqs(%d)\n", __func__, mb_ddr_1d[0].enableddqs);
-		ERROR("Value must be 0 < enableddqs < userinputbasic.numactivedbytedfi0 * 8.\n");
+		ERROR("%s %d", __func__, __LINE__);
+		VERBOSE("%s enableddqs(%d)\n", __func__, mb_ddr_1d[0].enableddqs);
+		VERBOSE("Value must be 0 < enableddqs < userinputbasic.numactivedbytedfi0 * 8.\n");
 	}
 
 	if (dbytenumber < 8) {
@@ -71,15 +74,17 @@ int ddrphy_phyinit_isdbytedisabled(int dbytenumber)
 #elif STM32MP_LPDDR4_TYPE
 	if (mb_ddr_1d[0].enableddqscha < 1 ||
 	    mb_ddr_1d[0].enableddqscha > 8 * userinputbasic.numactivedbytedfi0) {
-		ERROR("%s enableddqscha(%d)\n", __func__, mb_ddr_1d[0].enableddqscha);
-		ERROR("Value must be 0 < enableddqscha < userinputbasic.numactivedbytedfi0 * 8.\n");
+		ERROR("%s %d", __func__, __LINE__);
+		VERBOSE("%s enableddqscha(%d)\n", __func__, mb_ddr_1d[0].enableddqscha);
+		VERBOSE("Value must be 0 < enableddqscha < userinputbasic.numactivedbytedfi0*8\n");
 	}
 
 	if (userinputbasic.dfi1exists && userinputbasic.numactivedbytedfi1 > 0 &&
 	    (mb_ddr_1d[0].enableddqschb < 1 ||
 	     mb_ddr_1d[0].enableddqschb > 8 * userinputbasic.numactivedbytedfi1)) {
-		ERROR("%s enableddqschb(%d)\n", __func__, mb_ddr_1d[0].enableddqschb);
-		ERROR("Value must be 0 < enableddqschb < userinputbasic.numactivedbytedfi1 * 8.\n");
+		ERROR("%s %d", __func__, __LINE__);
+		VERBOSE("%s enableddqschb(%d)\n", __func__, mb_ddr_1d[0].enableddqschb);
+		VERBOSE("Value must be 0 < enableddqschb < userinputbasic.numactivedbytedfi1*8\n");
 	}
 #endif /* STM32MP_LPDDR4_TYPE */
 
diff --git a/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_mapdrvstren.c b/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_mapdrvstren.c
index 41d11e7b1..5694d1142 100644
--- a/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_mapdrvstren.c
+++ b/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_mapdrvstren.c
@@ -270,13 +270,15 @@ int ddrphy_phyinit_mapdrvstren(int drvstren_ohm, enum drvtype targetcsr)
 		} else if (drvstren_ohm == 20) {
 			stren_setting = 0x1F;
 		} else {
-			ERROR("%s userinputadvanced.atximpedance %d Ohms value is not valid.\n",
+			ERROR("%s %d", __func__, __LINE__);
+			VERBOSE("%s userinputadvanced.atximpedance %d Ohms value is not valid.\n",
 			      __func__, drvstren_ohm);
-			ERROR("Valid values are: 20, 24, 30, 40, 60 and 120 Ohms.\n");
+			VERBOSE("Valid values are: 20, 24, 30, 40, 60 and 120 Ohms.\n");
 		}
 
 	} else {
-		ERROR("targetcsr = %u, in %s, is not valid.\n",
+		ERROR("%s %d", __func__, __LINE__);
+		VERBOSE("targetcsr = %u, in %s, is not valid.\n",
 		      targetcsr, __func__);
 	}
 
diff --git a/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_reginterface.c b/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_reginterface.c
index 3d10529a0..ffa7bfc76 100644
--- a/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_reginterface.c
+++ b/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_reginterface.c
@@ -78,9 +78,10 @@ int ddrphy_phyinit_trackreg(uint32_t adr)
 
 	/* Register not found, so add it. */
 	if (numregsaved > MAX_NUM_RET_REGS) {
-		ERROR("[ddrphy_phyinit_reginterface:ddrphy_phyinit_trackreg]\n");
-		ERROR("Max Number of Restore Registers reached: %d.\n", numregsaved);
-		ERROR("Please recompile PhyInit with larger MAX_NUM_RET_REG value.\n");
+		ERROR("numregsaved > MAX_NUM_RET_REGS\n");
+		VERBOSE("[ddrphy_phyinit_reginterface:ddrphy_phyinit_trackreg]\n");
+		VERBOSE("Max Number of Restore Registers reached: %d.\n", numregsaved);
+		VERBOSE("Please recompile PhyInit with larger MAX_NUM_RET_REG value.\n");
 		return -1;
 	}
 
-- 
2.47.2

